### PR TITLE
Resolve conflicts between PR #152 and branch master

### PR DESCRIPTION
**Pull request by dragon540**

### CODE REVIEW

**Comment by klingaard** [comment]
:thinking:  Shobhit, how did you do the merge?  There shouldn't be this many files changed...

**Comment by dragon540** [comment]
@klingaard, I have mistakenly committed .idea and cmake-debug too, removing them should decrease the number of files changed to ~45 files. To do the merge, I copied the branch on PR #152 under my name, and merged master branch into it after resolving conflicts.

**Review by jeffnye-gh — CHANGES_REQUESTED** [review]
please remove the clion (.idea etc) and .cmake files from this PR and push again
also the cmake-build-debug/CMakeCache.txt cache files should not be part of the PR. I assume these are also part of clion.

you should consider adding these files to the toplevel .gitignore for future

**Comment by jeffnye-gh** [comment]
@Shobhit. I have a question, looking at Dans code the dcache operates on credits. This merge seems to revert to acks in the dcache, is this correct? and is it intentional?  

sparta::DataInPort<uint32_t> in_l2cache_credits_
vs.
sparta::DataInPort<uint32_t> in_l2cache_ack_

**Comment by dragon540** [comment]
removed extra files and folders (.idea, cmake-build-debug, etc) from the PR

**Review by jeffnye-gh — CHANGES_REQUESTED** [review]
Minor changes requested, remove some generated files.

File: test/core/l2cache/Testing/Temporary/LastTest.log [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Start testing: Sep 23 23:54 IST
+----------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (dragon540) [reply test/core/l2cache/Testing/Temporary/LastTest.log]: Removed generated log files

**Comment by dragon540** [comment]
Removed generated log files

**Review by jeffnye-gh — APPROVED** [review]

File: stf_lib
Comment (klingaard) [inline stf_lib:1]: Did you intend to update stflib?

**Review by klingaard — APPROVED** [review]
This is good work.  Before merging, I'd like to make sure moving stflib was intended.

**Comment by dragon540** [comment]
I was facing some issues with the version of stflib which was there in dan's PR, therefore updated it. However, the version of stflib which is in upstream branch works too. I have updated the stflib submodule to keep it consistent with the present version in olympia.

**Pull request merged**